src/Controller/ExerciseController.php

```diff
@@ -5,36 +5,45 @@
 use App\Entity\Exercise;
 use App\Form\ExerciseType;
 use App\Repository\ExerciseRepository;
+use Knp\Component\Pager\PaginatorInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\User\UserInterface;
 
 /**
  * @Route("/exercise")
  */
 class ExerciseController extends BaseController
 {
     /**
-     * @Route("/", name="exercise_index", methods={"GET"})
+     * @Route("/list/{page}/", defaults={"page"=1}, name="exercise_index", methods={"GET"})
      */
-    public function index(ExerciseRepository $exerciseRepository): Response
+    public function index(ExerciseRepository $exerciseRepository, UserInterface $user, PaginatorInterface $paginator, $page): Response
     {
+        $pagination = $paginator->paginate(
+            $exerciseRepository->findListForPagination($user->getId()),
+            $page,
+            $this::ITEMS_PER_PAGE
+        );
+
         return $this->render('exercise/index.html.twig', [
-            'exercises' => $exerciseRepository->findAll(),
+            'pagination' => $pagination,
         ]);
     }
 
     /**
      * @Route("/new", name="exercise_new", methods={"GET","POST"})
      */
-    public function new(Request $request): Response
+    public function new(Request $request, UserInterface $user): Response
     {
         $exercise = new Exercise();
         $form = $this->createForm(ExerciseType::class, $exercise);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $exercise = $form->getData();
+            $exercise->setUser($user);
```

Comment:
@daria-sieroshtan 
You should probably use [Blameable](https://symfony.com/doc/current/bundles/StofDoctrineExtensionsBundle/advanced.html)